mindsdb/integrations/handlers/litellm_handler/litellm_handler.py

```diff
@@ -0,0 +1,163 @@
+import ast
+from typing import Dict, Optional, List
+
+import pandas as pd
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities import log
+
+from mindsdb.integrations.handlers.litellm_handler.settings import CompletionParameters
+
+from litellm import completion, batch_completion
+
+# these require no additional arguments
+
+logger = log.getLogger(__name__)
+
+
+# todo add support for multiple api_keys in create engine. i.e. pass in keys for openai, anthropic, etc.
+
+
+class LiteLLMHandler(BaseMLEngine):
+    """
+    LiteLLMHandler is a MindsDB handler for litellm - https://docs.litellm.ai/docs/
+    """
+
+    name = "litellm"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.generative = True
+
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+        if "using" not in args:
+            raise Exception(
+                "Litellm engine requires a USING clause. See settings.py for more info on supported args."
+            )
+
+    def create(
+        self,
+        target: str,
+        df: pd.DataFrame = None,
+        args: Optional[Dict] = None,
+    ):
+        """
+        Dispatch is validating args and storing args in model_storage
+        """
+        # get api key from user input on create ML_ENGINE or create MODEL
+        input_args = args["using"]
+
+        # get api key from engine_storage
+        ml_engine_args = self.engine_storage.get_connection_args()
+
+        # check engine_storage for api_key
+        input_args.update({k: v for k, v in ml_engine_args.items()})
+
+        # validate args
+        export_args = CompletionParameters(**input_args).dict()
+
+        # store args
+        self.model_storage.json_set("args", export_args)
+
+    def update(self, args) -> None:
```

Comment:
In which context is this used? Doesn't seem to match any parent class methods, maybe `update_engine`?

Reply:
I hallucinated.. I removed this, its not needed